File: Easy/jewelStone/jewelStone.java
import java.util.Scanner;

public class jewelStone{
	public static void main(String[] args) {
		int T,count,note;
		Scanner scan = new Scanner(System.in);

		while(true){
			T = scan.nextInt();

			if(T<=0 || T>100){
				System.out.println("T must be 1 <= T <= 100");
			}else{
				break;
			}
		}

		String cut;

		while(T>0){
			String J = scan.next();
			String S = scan.next();

			if(J.length()>100){
				cut = J.substring(0,100);
				J = cut;
			}

			if(S.length()>100){
				cut = S.substring(0,100);
				S = cut;
			}

			count = 0;

			for(int i=0;i<S.length();i++){
				note = 0;
				for(int j=0;j<J.length();j++){
					if(S.charAt(i)==J.charAt(j) && note==0){
						count++;
						note++;
						break;
					}
				}
			}

			System.out.println(count);

			T--;
		}
	}
}

File: Easy/jewelStone/jewelStone.c
#include <stdio.h>
#include <string.h>

int main(void){
	int T,count,note;
	char J[100];
	char S[100];

	while(1){
		scanf("%d",&T);

		if(T<=0 || T>100){
			printf("T must be 1 <= T <= 100\n");
		}else{
			break;
		}
	}

	while(T>0){
		scanf("%s",J);
		scanf("%s",S);

		count = 0;

		for(int i=0;i<strlen(S);i++){
			note = 0;
			for(int j=0;j<strlen(J);j++){
				if(S[i]==J[j] && note==0){
					count++;
					note++;
					break;
				}

			}
		}

		printf("%d\n",count);

		T--;
	}

	return 0;
}